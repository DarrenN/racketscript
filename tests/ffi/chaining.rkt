#lang racket/base

(require racketscript/ffi)


(define console ($ 'global 'console))

(define (fun x)
  (if (zero? x)
    ($/obj [name      "Vishesh"]
           [location  "Boston"])
    ($/obj [onfoo     (λ (n)
                        ($$ console.log "Called foo: " n)
                        (fun n))]
           [onbar     (λ (n)
                        ($$ console.log "Called bar " n)
                        (fun n))])))

($ ($> (fun 10)
       (onfoo 10)
       (onbar 20)
       (onfoo 30)
       (onbar 40)
       (onbar 50)
       (onfoo 0))
   'name)
