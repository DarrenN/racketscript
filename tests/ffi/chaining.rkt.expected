Called foo:  10
Called bar:  20
Called foo:  30
Called bar:  40
Called bar:  50
Called foo:  0
Vishesh
